packages/plugin-core/src/test/suite-integ/NoteLookupCommand.test.ts

```diff
@@ -2225,619 +2114,367 @@ suite("NoteLookupCommand", function () {
+        runLegacyMultiWorkspaceTest({
+          ctx,
+          preSetupHook: async ({ wsRoot, vaults }) => {
+            await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+          },
+          onInit: async ({ wsRoot, vaults, engine }) => {
+            const { selectedText, cmd } = await prepareCommandFunc({
+              vaults,
+              engine,
+              noteType: LookupNoteTypeEnum.journal,
+            });
 
-    test("scratch and selection2link both applied", (done) => {
-      runLegacyMultiWorkspaceTest({
-        ctx,
-        preSetupHook: async ({ wsRoot, vaults }) => {
-          await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
-        },
-        onInit: async ({ vaults, engine }) => {
-          const { controller } = await prepareCommandFunc({ vaults, engine });
+            const dateFormat = ConfigUtils.getJournal(engine.config).dateFormat;
+            const today = Time.now().toFormat(dateFormat);
+            const newNote = NoteUtils.getNoteOrThrow({
+              fname: `foo.journal.${today}`,
+              notes: engine.notes,
+              vault: vaults[0],
+              wsRoot,
+            });
 
-          const { scratchBtn } = getNoteTypeButtons(
-            controller.quickpick.buttons
-          );
+            expect(newNote.body.trim()).toEqual(selectedText);
 
-          const { selection2linkBtn } = getSelectionTypeButtons(
-            controller.quickpick.buttons
-          );
+            cmd.cleanUp();
+            done();
+          },
+        });
+      });
 
-          expect(scratchBtn.pressed).toBeTruthy();
-          expect(selection2linkBtn.pressed).toBeTruthy();
+      test("scratch + selectionExtract both applied", (done) => {
+        runLegacyMultiWorkspaceTest({
+          ctx,
+          preSetupHook: async ({ wsRoot, vaults }) => {
+            await ENGINE_HOOKS.setupBasic({ wsRoot, vaults });
+          },
+          onInit: async ({ wsRoot, vaults, engine }) => {
+            const { cmdOut, selectedText, cmd } = await prepareCommandFunc({
+              vaults,
+              engine,
+              noteType: LookupNoteTypeEnum.scratch,
+            });
 
-          const todayFormatted = getTodayInScratchDateFormat();
-          const quickpickValue = controller.quickpick.value;
-          expect(
-            quickpickValue.startsWith(`scratch.${todayFormatted}`)
-          ).toBeTruthy();
-          expect(quickpickValue.endsWith(".foo-body")).toBeTruthy();
+            const newNote = NoteUtils.getNoteOrThrow({
+              fname: cmdOut!.quickpick.value,
+              notes: engine.notes,
+              vault: vaults[0],
+              wsRoot,
+            });
+            expect(newNote.body.trim()).toEqual(selectedText);
 
-          done();
-        },
+            cmd.cleanUp();
+            done();
+          },
+        });
       });
     });
 
-    test("toggling scratch modifier off will only leave selection2link applied", (done) => {
```

Comment:
@hikchoi - there's a series of tests here around "toggling * only leaves selection2link applied".  Do you remember the rationale behind this behavior or what the scenario was about?  

I can re-add corresponding tests in the controller test, but I had to remove these particular ones because `onTriggerButton()` is no longer exposed.

Reply:
these are tests to verify that toggling a modifier button of different type back and forth will revert back the quickpick values to the original value.

e.g.)

start with with selection2link and scratch modifier applied. At this point the quickpick value would be something like `scratch.yyyy.mm.dd.some-selection`

I was testing here that toggling scratch modifier off would leave the quickpick value as `{current-note}.some-selection`.

These were added because there was a regression during the lookup v3 migration where quickpick values were not being restored or becoming some garbage value.

I went a little extra with covering every single combination here, so it may be a good idea to trim them down if you find them a bit redundant.